exercises/custom-set/example.rs

```diff
@@ -1,149 +1,95 @@
+            self.collection.push(element)
         }
-        false
     }
 
-    pub fn remove(&mut self, needle: &T) -> bool {
-        let mut index: Option<usize> = None;
-        for (i, element) in self.iter().enumerate() {
-            if needle == element {
-                index = Some(i);
-                break;
-            }
-        }
-        match index {
-            Some(index) => {
-                self.elements.remove(index);
-                true
-            }
-            None => false
-        }
-    }
-    
-    pub fn difference(&self, other: &CustomSet<T>) -> CustomSet<T> {
-        let mut difference: CustomSet<T> = CustomSet::new();
-        for element in self.elements.iter() {
-            if !other.contains(element) {
-                difference.insert(element.clone());
-            }
-        }
-        difference
+    pub fn delete(&mut self, element: T) {
+        self.collection.retain(|x| x != &element)
     }
 
-    pub fn intersection(&self, other: &CustomSet<T>) -> CustomSet<T> {
-        let mut intersection: CustomSet<T> = CustomSet::new();
-        for element in self.elements.iter() {
-            if other.contains(element) {
-                intersection.insert(element.clone());
-            }
-        }
-        intersection
+    pub fn contains(&self, other: &T) -> bool {
+        self.collection.contains(other)
     }
 
-    pub fn union(&self, other: &CustomSet<T>) -> CustomSet<T> {
-        let mut union = CustomSet::new();
-        for element in self.elements.iter().chain(other.elements.iter()) {
-            union.insert(element.clone());
-        }
-        union
+    pub fn is_empty(&self) -> bool {
+        self.collection.is_empty()
     }
 
-    pub fn is_disjoint(&self, other: &CustomSet<T>) -> bool {
-        for element in self.elements.iter() {
-            if other.contains(element) {
-                return false;
-            }
-        }
-        true
+    pub fn size(&self) -> usize {
+        self.collection.len()
     }
 
-    pub fn is_subset(&self, other: &CustomSet<T>) -> bool {
-        for element in self.elements.iter() {
-            if !other.contains(element) {
-                return false;
-            }
-        }
-        true
-    }
-    
-    pub fn is_superset(&self, other: &CustomSet<T>) -> bool {
-        other.is_subset(self)
+    pub fn is_subset(&self, other: &Self) -> bool {
+        self.collection.iter().all(|x| other.contains(x))
     }
 
-    pub fn clear(&mut self) {
-        self.elements.clear();
+    pub fn is_disjoint(&self, other: &Self) -> bool {
+        !self.collection.iter().any(|x| other.contains(x))
     }
 
-    pub fn iter(&self) -> Iter<T>{
-        Iter { iter: self.elements.iter() }
+    pub fn intersection(&self, other: &Self) -> CustomSet<T> {
+        CustomSet::new(self.collection
+                           .iter()
+                           .filter(|c| other.contains(c))
+                           .map(|c| c.clone())
```

Comment:
Seems like a place for https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.cloned